compiler/env/FEBase_t.hpp

```diff
@@ -52,25 +53,28 @@ FEBase<Derived>::allocateCodeMemory(TR::Compilation *comp, uint32_t warmCodeSize
 
    uint8_t *warmCode = codeCacheManager().allocateCodeMemory(warmCodeSize, coldCodeSize, &codeCache,
                                                              coldCode, false, isMethodHeaderNeeded);
-   if (warmCode == NULL)
-      {
-      comp->setErrorCode(jitConfig()->isCodeCacheFull() ?
-                         COMPILATION_CODE_MEMORY_EXHAUSTED :
-                         COMPILATION_ALL_CODE_CACHES_RESERVED);
-      }
+
    if (codeCache != comp->getCurrentCodeCache())
       {
       // Either we didn't get a code cache, or the one we get should be reserved
       TR_ASSERT(!codeCache || codeCache->isReserved(), "Substitute code cache isn't marked as reserved");
       comp->setAotMethodCodeStart(warmCode);
       switchCodeCache(codeCache);
       }
-   if ( comp->getErrorCode() != COMPILATION_SUCCEEDED )
+
+   if (warmCode == NULL)
       {
-      outOfMemory(comp, "code");
+      if (jitConfig()->isCodeCacheFull())
+         {
+         throw TR::CodeCacheError();
+         }
+      else
+         {
+         throw TR::RecoverableCodeCacheError();
+         }
```

Comment:
In the OMR project the vlog doesn't print anything special, it would've just printed out "code". Now it will print either "Code Cache Error" or "Recoverable Code Cache Error".